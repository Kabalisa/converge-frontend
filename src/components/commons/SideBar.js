import React, { Component } from 'react';
import propTypes from 'prop-types';
import home from '../../assets/images/Home_active (2).svg';
import meetings from '../../assets/images/Meetings_inactive.svg';
import insights from '../../assets/images/Insights_inactive.svg';
import setup from '../../assets/images/Setup_inactive.svg';
import device from '../../assets/images/Tablet_inactive.svg';
import resources from '../../assets/images/Resources_inactive.svg';
import people from '../../assets/images/People_inactive.svg';
import rooms from '../../assets/images/Rooms_inactive.svg';
import '../../assets/styles/sidebar.scss';

// eslint-disable-next-line react/prefer-stateless-function
class SideBar extends Component {
  // eslint-disable-next-line no-useless-constructor
  constructor(props) {
    super(props);
    this.state = { name: 'home' };
  }
  handleClick = ({ target: { name } }) => {
    this.setState({ name });
    this.props.setContent(name);
  }
  render() {
    const { name } = this.state;
    return (
      <div className="sidebar">
        <p className="overview-text">OVERVIEW</p>
        <a
          name="home"
          className={name === 'home' ? 'active' : 'home'}
          href="#home"
          onClick={e => this.handleClick(e)}
        >
          <img src={home} alt="" />
          Home
        </a>
        <a
          name="meetings"
          className={name === 'meetings' ? 'active' : 'meetings'}
          href="#meetings"
          onClick={e => this.handleClick(e)}
        >
          <img src={meetings} alt="" />
          Meetings
        </a>
        <a
          name="insights"
          className={name === 'insights' ? 'active' : 'insights'}
          href="#insights"
          onClick={e => this.handleClick(e)}
        >
          <img src={insights} alt="" />
          Insights
        </a>
        <a
          name="setup"
          className={name === 'setup' ? 'active' : 'setup'}
          href="#setup"
          onClick={e => this.handleClick(e)}
        >
          <img src={setup} alt="" />
          Setup
        </a>
        <p className="overview-text">MANAGE</p>
        <a href="#devices" onClick={() => this.props.setContent('devices')}>
          <img src={device} alt="" />
          devices
        </a>
        <a href="#resources" onClick={() => this.props.setContent('resources')}>
          <img src={resources} alt="" />
          Resources
        </a>
        <a href="#people" onClick={() => this.props.setContent('people')}>
          <img src={people} alt="" />
          People
        </a>
        <a href="#rooms" onClick={() => this.props.setContent('rooms')}>
          <img src={rooms} alt="" />
          Rooms
        </a>
      </div>
    );
  }
}
SideBar.propTypes = {
  setContent: propTypes.func.isRequired,
};

export default SideBar;
