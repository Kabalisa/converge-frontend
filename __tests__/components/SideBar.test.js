import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import SideBar from '../../src/components/SideBar';

describe('SideBar Component', () => {
  const wrapper = shallow(<SideBar />);

  it('Should render correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SideBar />, div);
  });
  it('renders correctly in memory', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
