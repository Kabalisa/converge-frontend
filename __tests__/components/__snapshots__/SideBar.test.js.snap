// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component renders correctly in memory 1`] = `
<div
  className="side-bar"
>
  SideBar
</div>
`;
